src/components/ConfigProvider/FeatureFlagProvider.tsx

```diff
@@ -0,0 +1,49 @@
+'use client';
+
+import React, { createContext, FC, useContext, useMemo } from 'react';
+
+export interface FeatureFlags {
+  panelLazyLoadContent?: boolean;
+}
+
+export interface FeatureFlagContextProps {
+  featureFlags?: FeatureFlags;
+  children?: React.ReactNode;
+}
+
+const FeatureFlagContext = createContext<FeatureFlags>(undefined);
+
+const defaultFeatureFlagValues: FeatureFlags = {
+  panelLazyLoadContent: false,
+};
```

Comment:
Should be able to initialize with the default values and remove the spreading of defaults to inner contexts.
```suggestion
const FeatureFlagContext = createContext<FeatureFlags>({
  panelLazyLoadContent: false,
});
```

Reply:
I ended up using the default values below, in cases where there's no config available